core/math/quaternion.h

```diff
@@ -141,16 +141,26 @@ struct [[nodiscard]] Quaternion {
 	}
 
 	Quaternion(const Vector3 &p_v0, const Vector3 &p_v1) { // Shortest arc.
-		Vector3 c = p_v0.cross(p_v1);
+#ifdef MATH_CHECKS
+		ERR_FAIL_COND_MSG(p_v0.is_zero_approx() || p_v1.is_zero_approx(), "The vectors must not be zero.");
+#endif
+		Vector3 n0 = p_v0.normalized();
+		Vector3 n1 = p_v1.normalized();
+
+		real_t d = n0.dot(n1);
+		real_t n = 1.0f - (real_t)CMP_EPSILON;
+		if (d > n) {
+			return;
+		}
 
-		if (c.is_zero_approx()) {
-			Vector3 axis = p_v0.get_any_perpendicular();
+		if (d < -n) {
```

Comment:
I am not sure of the strictness of `-(1.0f - (real_t)CMP_EPSILON)` vs `-1.0f + (real_t)CMP_EPSILON`. Is it fine to reuse the `n`?

Reply:
Yeah it's fine afaik. You could make it a `constexp` too if you wanted to make more clear, but the compiler should realize it's a `const`.

Compiler should add two separate consts (one for positive and one for negative). Or rather I don't see why it wouldn't.

Reply:
ah const sounds good.